src/swap/components/SwapToast.tsx

```diff
@@ -0,0 +1,94 @@
+import { useCallback, useEffect, useMemo } from 'react';
+import { closeSvg } from '../../internal/svg/closeSvg';
+import { background, cn, color, text } from '../../styles/theme';
+
+import { useAccount } from 'wagmi';
+import { successSvg } from '../../internal/svg/successSvg';
+import { getToastPosition } from '../../internal/utils/getToastPosition';
+import { getChainExplorer } from '../../network/getChainExplorer';
+import type { SwapToastReact } from '../types';
+import { useSwapContext } from './SwapProvider';
+
+export function SwapToast({
+  className,
+  durationMs = 3000,
+  position = 'bottom-center',
+}: SwapToastReact) {
+  const {
+    isToastVisible,
+    setIsToastVisible,
+    setTransactionHash,
+    transactionHash,
+  } = useSwapContext();
+
+  const { chainId } = useAccount();
+
+  const chainExplorer = getChainExplorer(chainId);
+
+  const closeToast = useCallback(() => {
+    setIsToastVisible?.(false);
+  }, [setIsToastVisible]);
+
+  const positionClass = useMemo(() => {
+    return getToastPosition(position);
+  }, [position]);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (isToastVisible) {
+        setIsToastVisible?.(false);
+        setTransactionHash?.('');
+      }
+    }, durationMs);
+
+    return () => {
+      if (timer) {
+        clearTimeout(timer);
+      }
+    };
+  }, [durationMs, isToastVisible, setIsToastVisible, setTransactionHash]);
+
+  if (!isToastVisible) {
+    return null;
+  }
+
+  return (
+    <div
+      className={cn(
+        background.default,
+        'flex animate-enter items-center justify-between rounded-lg',
+        'p-2 shadow-[0px_8px_24px_0px_rgba(0,0,0,0.12)]',
+        '-translate-x-2/4 fixed z-20',
```

Comment:
the animate in from the right seems weird, also, should this be pegged to the swap component vs. the window?

Reply:
so will investigate but it appears like it was an existing issue with how we style our toasts